allennlp/training/trainer.py

```diff
@@ -988,13 +1001,11 @@ def from_params(cls,
         validation_metric = params.pop("validation_metric", "-loss")
         shuffle = params.pop_bool("shuffle", True)
         num_epochs = params.pop_int("num_epochs", 20)
-        cuda_device = params.pop_int("cuda_device", -1)
+        cuda_device = parse_cuda_device(params.pop("cuda_device", -1))
         grad_norm = params.pop_float("grad_norm", None)
         grad_clipping = params.pop_float("grad_clipping", None)
         lr_scheduler_params = params.pop("learning_rate_scheduler", None)
 
-        if cuda_device >= 0:
-            model = model.cuda(cuda_device)
```

Comment:
@vidurj, any chance you could comment on why this block was removed? It ended up causing problems when a particular optimizer was used. See https://github.com/allenai/allennlp/issues/2189. Reversion here: https://github.com/allenai/allennlp/pull/2190.

Was this just a cleanup or something deeper? (Uses GPU memory sooner, notably, which could cause issues...) Basically I want to check that the reversion is safe in a multi-GPU setting. :) Thanks!

Reply:
I had no reason to remove this block. I suspect this was inherited from the PR I was adding tests to.